pkg/datapath/linux/node_addressing.go

```diff
@@ -28,20 +28,11 @@ func listLocalAddresses(family int) ([]net.IP, error) {
 		return nil, err
 	}
 
-	for _, addr := range addrs {
-		if addr.Scope > option.Config.AddressScopeMax {
-			continue
-		}
-		if ip.ListContainsIP(ipsToExclude, addr.IP) {
-			continue
-		}
-		if addr.IP.IsLoopback() || addr.IP.IsLinkLocalUnicast() {
-			continue
-		}
-
-		addresses = append(addresses, addr.IP)
-	}
+	filteredIPs := filterLocalAddresses(addrs, ipsToExclude, option.Config.AddressScopeMax)
+	addresses = append(addresses, filteredIPs...)
 
+	// If AddressScopeMax is a scope more broad (numerically less than) than SCOPE_LINK then include
+	// all addresses at SCOPE_LINK which are assigned to the Cilium host device.
 	if option.Config.AddressScopeMax < int(netlink.SCOPE_LINK) {
```

Comment:
@borkmann Should this be "<=" rather than "<"? If our max scope is `SCOPE_LINK`, then I think we should include these addresses.

Reply:
Right, makes sense